src/sql/azdata.proposed.d.ts

```diff
@@ -3918,6 +3918,31 @@ declare module 'azdata' {
 	 * Namespace for interacting with query editor
 	*/
 	export namespace queryeditor {
+		export type QueryEvent =
+			| 'queryStart'
+			| 'queryStop'
+			| 'executionPlan';
+
+		export interface QueryEventListener {
+			onQueryEvent(type: QueryEvent, document: queryeditor.QueryDocument, args: any);
+		}
+
+		// new extensibility interfaces
+		export interface QueryDocument {
+			providerId: string;
+
+			uri: string;
+
+			// get the document's execution options
+			getOptions(): Map<string, string>;
+
+			// set the document's execution options
+			setOptions(options: Map<string, string>): void;
+
+			// tab content is build using the modelview UI builder APIs
+			// probably should rename DialogTab class since it is useful outside dialogs
+			createQueryTab(tab: window.DialogTab): void;
```

Comment:
It seems to me AddQueryTab makes more sense here.

Reply:
`create` seems to be the verb we're using for this type of action generally.  We can discuss whether to change globally in a design review going from proposed to stable, but for now I want to keep consistent with what already in the API.